### PR TITLE
1354 display more useful error message on login for nonvalidated emai…

### PR DESCRIPTION
**Pull request by avaek**
Addresses: https://github.com/beyondessential/tupaia-backlog/issues/1354

To be honest I'm not sure what a reducer is - making this one line change to this file seemed to fix the issue (from the perspective of the front end) but I'm not sure if I have to change things in other places too (there seems to be something related to this in `actions.js` but I'm not sure what the relationship between the two files is & I'm not sure if I had to make a change to that file too since just changing this one file seemed to solve the issue)

Basically this change means that the error message that is displayed to the front end comes from the error supplied by the `auth` package (rather than the hardcoded string that it used to display)

### CODE REVIEW

**Comment by Alastair-L** [comment]
Hey @avaek, I do like this change! However, I have 2 possible improvements to it (but first a quick background on redux):

**Reducers** and **actions** are both concepts from redux, which is the framework that we use to control all the global storage in the frontend. 

In redux, the ONLY way that **state** can be changed is through the specific reducer that controls that piece of state. In this case, the piece of state `loginFailedMessage` is controlled by the `authentication` reducer (and you would access it with `authentication.loginFailedMessage`).

The only time reducers themselves are called is when **actions** are **dispatched**. In this case, there are 2 actions that can cause  `loginFailedMessage` to change: 
"OPEN_USER_DIALOG" and "ATTEMPT_LOGIN", which set it to `null`
"FETCH_LOGIN_ERROR". which you are familiar with, and
"FETCH_EMAIL_VERIFY_ERROR", which is called **after** clicking on the verify email link and is therefore no use to us.

On the action side of things, we can see the action `fetchUserLoginError` has the type 'FETCH_LOGIN_ERROR', and therefore will trigger our case whenever it is called.

Note: WHENEVER an action is dispatched, ALL reducers are called. It is the switch on action.type that determines which ones change (not relevant here, just might help understanding what is going on).

If you don't have [redux dev tools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd) yet, I would highly recommend getting them and having a little explore.

More in depth tutorial about redux [here](https://www.tutorialspoint.com/redux/redux_quick_guide.htm)



Now on to my suggestions:
1.  The FETCH_LOGIN_ERROR action already has an `errorMessage` key, which at the moment is not used. I would suggest removing that key (although it isn't really your fault, just a small code cleanup)
2. I would also like to see a default message in case `errors` is not defined or it has no key called `error`. Perhaps "Unknown error during login", or something.
3. I noticed that `errors` is not actually an array, although it is initialized to be one (see the start of `authentication`) and it's name suggests that it should be. I believe that we could solve this as part of this PR as well (this one is optional for me, but it wouldn't hurt to at least add a comment noting it).

**Comment by avaek** [comment]
@Alastair-L - many thanks for that thorough review!

I think your idea about solving this in `sagas.js` is a really great idea because in addition to solving this card it also solves another one for free ( https://github.com/beyondessential/tupaia-backlog/issues/1355 )

I reverted my previous change but still performed the code cleanup you mentioned since I think this is still okay, is that still the right thing to do?

Also I'm not quite sure why the line in `sagas.js` was written the way it was but I just changed it from `details` to `error` like you mentioned because I assume it was written this way for a specific reason (I mostly don't understand why it was written `if (errorMessage.details && errorMessage.details === 'Email address not yet verified')` instead of just writing `if (errorMessage.details === 'Email address not yet verified')` but I'm sure there's a reason for this)

**Review by Alastair-L — CHANGES_REQUESTED** [review]
Nice work, and absolutely that `errorMessage` key is still unused!

I expect that there is probably a more robust way of handling this (maybe using error codes rather that the message) and explicitly handling more possible errors such as "User not found" (search `getAuthenticatedUser` in the auth package). However, I think that's out of scope for this issue.

Just that one small suggestion, then looks good 👍

File: packages/web-frontend/src/sagas.js [hunk 1]
```diff
@@ -243,7 +243,7 @@ function* attemptUserLogin(action) {
     yield put(findLoggedIn(LOGIN_TYPES.MANUAL, response.emailVerified));
   } catch (error) {
     const errorMessage = error.response ? yield error.response.json() : {};
-    if (errorMessage.details && errorMessage.details === 'Email address not yet verified') {
+    if (errorMessage.error && errorMessage.error === 'Email address not yet verified') {
```
[the remark this answers is not in the document]
Comment (kael89) [reply packages/web-frontend/src/sagas.js:246]: `?.` will work, go for it! `errorMessage && errorMessage.error` is also fine

**Review by Alastair-L — APPROVED** [review]
LGTM! Nice work.
